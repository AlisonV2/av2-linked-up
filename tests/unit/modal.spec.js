import { shallowMount } from '@vue/test-utils';
import AppModal from '@/components/app/Modal';

/**
 * @module AppModalTest
 */
describe('AppModal.vue', () => {
  it('Modal should not be visible', () => {
    const component = shallowMount(AppModal, {
        props: {
            show: false,
            scrollable: false
        }
    });
    expect(component.find('.modal-dialog')).not.toBeVisible;
  });

  it('Modal should be visible', () => {
    const component = shallowMount(AppModal, {
        props: {
            show: true,
            scrollable: false
        }
    });
    expect(component.find('.modal-dialog')).toBeVisible;
  });
  it('should render correct contents', () => {
    const component = shallowMount(AppModal, {
        props: {
            show: true,
            scrollable: false
        }
    });
    component.get('.close-btn').trigger('click');
    expect(component.find('.modal-dialog')).not.toBeVisible;
  });
  it('Close event should be called', () => {
    const component = shallowMount(AppModal, {
        props: {
            show: true,
            scrollable: false
        }
    });
    component.get('.close-btn').trigger('click');
    expect(component.emitted().close).toHaveBeenCalled;
  });
  it('Handler method should be called', () => {
    const component = shallowMount(AppModal, {
        props: {
            show: true,
            scrollable: false
        }
    });
    component.get('.modal-backdrop').trigger('click');
    expect(component.emitted().onClickAway).toHaveBeenCalled;
  });
  it('Handler method should be called', () => {
    const component = shallowMount(AppModal, {
        props: {
            show: true,
            scrollable: false
        }
    });
    AppModal.unmounted.call(component.vm);
    expect(component.emitted().handler).toHaveBeenCalled;
  });
});