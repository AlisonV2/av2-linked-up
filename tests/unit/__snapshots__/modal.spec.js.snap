// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppModal.vue PreviewItem Snapshot 1`] = `
<div
  class="modal"
  style="display: block;"
>
  <div
    class="modal-dialog"
    style="z-index: 2000;"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <h5
          class="modal-title"
        >
          
          
        </h5>
        <span
          class="close-btn"
        >
          Ã—
        </span>
      </div>
      <div
        class="modal-body"
      >
        <p>
          
          
        </p>
      </div>
    </div>
  </div>
  <div
    class="modal-backdrop show"
  />
</div>
`;
